benchmarks/benchmark-from-entry-hash.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+const Log = require('../src/log')
+const EntryIO = require('../src/entry-io')
+const IPFS = require('ipfs')
+const IPFSRepo = require('ipfs-repo')
+const DatastoreLevel = require('datastore-level')
+
+// State
+let ipfs
+let log
+
+// Metrics
+let totalQueries = 0
+let seconds = 0
+let queriesPerSecond = 0
+let lastTenSeconds = 0
+let total = 0
+
+let run = (() => {
+  console.log('Starting benchmark...')
+
+  const repoConf = {
+    storageBackends: {
+      blocks: DatastoreLevel,
+    },
+  }
+
+  ipfs = new IPFS({
+    repo: new IPFSRepo('./ipfs-log-benchmarks/fromEntryHash/ipfs', repoConf),
+    start: false,
+    EXPERIMENTAL: {
+      pubsub: false,
+      sharding: false,
+      dht: false,
+    },
+  })
+
+  ipfs.on('error', (err) => {
+    console.error(err)
+  })
+
+  ipfs.on('ready', async () => {
+    // Create a log
+    log = new Log(ipfs, 'A')
+
+    const count = parseInt(process.argv[2]) || 50000
+    const refCount = 64
+    const concurrency = 128
+    const delay = 0
+
+    console.log("Creating a log...")
+
+    const st = new Date().getTime()
+
+    try {
+    for (let i = 1; i < count + 1; i ++) {
+      await log.append('hello' + i, refCount)
+      process.stdout.write("\rWriting " + i + " / " + count)
+    }
+    const dt1 = new Date().getTime()
+      process.stdout.write(" (" + (dt1 - st) + " ms)\n")
+    } catch (e) {
+      console.log(e)
+    }
+
+
+    const onDataUpdated = (hash, entry, resultLength, result, queue) => {
+      // totalQueries = resultLength
+      queriesPerSecond++
+      lastTenSeconds++
+      total = resultLength
+      process.stdout.write("\rLoading " + total + " / " + count)
+    }
+
+    const outputMetrics = () => {
+      // queriesPerSecond = total - queriesPerSecond
+      totalQueries = total - totalQueries
+      seconds++
+      if (seconds % 10 === 0) {
+        console.log(`--> Average of ${lastTenSeconds / 10} q/s in the last 10 seconds`)
+        if (lastTenSeconds === 0) throw new Error('Problems!')
+        lastTenSeconds = 0
+      }
+      console.log(`\n${queriesPerSecond} queries per second, ${totalQueries} queries in ${seconds} seconds (Entry count: ${total})`)
```

Comment:
If I read the code correctly, we're benchmarking "how long does it take to load a log" and as such we're not necessarily doing queries, but "entries per second". I feel it'd be worthwhile to make the output messages clearer on that and display something like "X entries loaded per second" etc.

What do you think?

If you feel that would improve the PR, we should also rename the variables used in the code to match the output, eg. `queriesPerSecond` -> `entriesLoadedPerSecond` or something similar.

Reply:
No brainer change - lazy of me to have not done it before 😁